src/com/github/kornilova203/matlab/resolveUtil.kt

```diff
@@ -44,7 +42,7 @@ fun PsiElement.findDeclaration(processor: PsiScopeProcessor, state: ResolveState
 }
 
 fun processDeclarations(parent: PsiElement, processor: PsiScopeProcessor, state: ResolveState, lastParent: PsiElement?, place: PsiElement): Boolean {
-    val inCurrentScope = place.inCurrentFunction(parent)
+    val inCurrentScope = place.inCurrentFunction(parent) && place.parent !is MatlabFunctionExpr
```

Comment:
In this case `inCurrentScope` has to be renamed to `inCurrentScopeAndNotAFunctionCall` but this would be too complicated.

With current implementation you cannot resolve function references that are not calls:
```matlab
fun<ref>

function fun
end
```

I would add boolean property `visibleBeforeDeclaration` to `MatlabDeclaration` that is `true` for functions and maybe global variables and classes (?)

Then remove line `if (inCurrentScope) return true` and pass boolean flag `declarationIsBeforeUsage` to `PsiElement.findDeclaration` function.